README.md
# GoogleMapsCross

A flutter package that allows you to create basic timelines on your flutter application. This is customizable and easy to plugin to your application. You will find the following features:

* Usage google maps with crossplatform

<image src="https://github.com/zapiti/google_maps_cross/blob/main/maps.png" height="300em"></image>

## Getting Started

You will need to add the following dependency in your `pubspec.yaml` file to download the depedency.

```dart
dependencies:
  flutter:
    sdk: flutter
  google_maps_cross: 1.0.0
```


## Usage

> import 'package:google_maps_cross/google_maps_cross.dart';

### WEB
   * In file index.html:
   
   
    <body id="app-container">
    <script src="https://maps.googleapis.com/maps/api/js?key=API_KEY"></script>
    <script src="main.dart.js?version=2.1.33" type="application/javascript"></script>
    </body>

    
### iOS
 * In file AppDelegate.swift:
 
    ```import UIKit
    import Flutter
    import GoogleMaps

    @UIApplicationMain
    @objc class AppDelegate: FlutterAppDelegate {
      override func application(
        _ application: UIApplication,
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
      ) -> Bool {
        GMSServices.provideAPIKey("API_KEY")
        GeneratedPluginRegistrant.register(with: self)
        return super.application(application, didFinishLaunchingWithOptions: launchOptions)
      }
    }
  
    
 ### Android
   * In file AndroidManifest.xml:
    
     ```<application
            android:label="example"
            android:icon="@mipmap/ic_launcher">
           <meta-data android:name="com.google.android.geo.API_KEY"
               android:value="API_KEY"/>
            <activity
                android:name=".MainActivity"
                android:launchMode="singleTop"
                android:theme="@style/LaunchTheme"
                android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
                android:hardwareAccelerated="true"
                android:windowSoftInputMode="adjustResize">

## Implementation

```
            GoogleMapCross(
                  key: key,
                  interactive: true,
                  onTap: (x) {},
                  initialPosition: GeoCoord(-28.4831, -47.3916),
                  initialZoom: 5.0,
                  markers: Set.from([
                    Marker(GeoCoord(-18.4435, -47.3415), onTap: (_) {
                      print("description3");
                    })
                  ]),
                ),
```

## Contributing

Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.



CHANGELOG.md
## 1.0.2 - release
* Suport flutter 3

## 1.0.1 - release
  * Update dependencies

## 1.0.0 - Get start
  * Start google_maps_cross


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:google_maps_cross/google_maps_cross.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter GOOGLE MAPS CROSS',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter GOOGLE MAPS CROSS'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  GlobalKey<GoogleMapStateBase> key = GlobalKey<GoogleMapStateBase>();

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Expanded(
              child: Container(
                margin: EdgeInsets.only(right: 10),
                width: MediaQuery.of(context).size.width,
                child: GoogleMapCross(
                  key: key,
                  interactive: true,
                  onTap: (x) {},
                  initialPosition: GeoCoord(-28.4831, -47.3916),
                  initialZoom: 5.0,
                  markers: Set.from([
                    Marker(GeoCoord(-18.4435, -47.3415), onTap: (_) {
                      print("description3");
                    })
                  ]),
                ),
              ),
            )
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          GoogleMapCross.of(key)?.addMarker(Marker(GeoCoord(-19.4831, -48.3916),
              icon: 'assets/images/truck_blue.png', onTap: (_) {
            print("description");
          }));

          GoogleMapCross.of(key)
              ?.addMarker(Marker(GeoCoord(-20.5855, -48.4015), onTap: (_) {
            print("description3");
          }));
        },
        tooltip: 'Increment',
        child: Icon(Icons.add_location),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
